docs/fsharp/introduction-to-functional-programming/index.md

```diff
@@ -1,8 +1,280 @@
+
+### Expressions
+
+Expressions are constructs that evaluate to a value. In contrast to statements, which perform an action, expressions can be thought of performing an action that gives back a value. Expressions are almost always used in favor of statements in functional programming.
+
+Consider the previous function, `addOne`. The body of `addOne` is an expression:
+
+```fsharp
+// 'x + 1' is an expression!
+let addOne x = x + 1
+```
+
+It is the result of this expression that defines the result type of the `addOne` function. For example, the expression that makes up this function could be changed to be a different type, such as a `string`:
+
+```fsharp
+let addOne x = x.ToString() + "1"
+```
+
+The signature of the function is now:
+
+```fsharp
+val addOne: x:'a -> string
+```
+
+Since any type in F# can have `ToString()` called on it, the type of `x` has been made generic (called [Automatic Generalization](../language-reference/generics/automatic-generalization.md)), and the resultant type is a `string`.
+
+Expressions are not just the bodies of functions. You can have expressions that produce a value you use elsewhere. A common one is `if`:
+
+```fsharp
+// Checks if 'x' is odd by using the mod operator
+let isOdd x = x % 2 <> 0
+
+let addOneIfOdd input =
+    let result =
+        if isOdd input then
+            input + 1
+        else
+            input
+
+    result
+```
+
+The `if` expression produces a value called `result`. Note that you could omit `result` entirely, making the `if` expression the body of the `addOneIfOdd` function. The key thing to remember about expressions is that they produce a value.
+
+There is a special type, `unit`, that is used when there is nothing to return. For example, consider this simple function:
+
+```fsharp
+let printString (str: string) =
+    printfn "String is: %s" s
+```
+
+The signature looks like this:
+
+```fsharp
+val printString: str:string -> unit
+```
+
+The `unit` type indicates that there is no actual value being returned. This is useful when you have a routine that must "do work" despite having no value to return as a result of that work.
+
+This is in sharp contrast to imperative programming, where the equivalent `if` construct is a statement, and producing values is often done with mutating variables. For example, in C#, the code might be written like this:
+
+```csharp
+bool IsOdd(int x) => x % 2 != 0;
+
+int AddOneIfOdd(int input)
+{
+    var result = input;
+
+    if (IsOdd(input))
+    {
+        result = input + 1;
+    }
+
+    return result;
+}
+```
+
+It's worth noting that C# and other C-style languages do suppor the ternary `?` expression, which allows for expression-based conditional programming.
```

Comment:
typo: "do suppor" -> "do support"
nit: `?` -> `?:` (not to mix with the null-conditional `?.` operator) or even explicitly: `cond ? x : y`